File: src-tauri/src/domain/entities/account.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "lowercase")]
pub enum AccountType {
    Offline,
    Microsoft,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MinecraftAccount {
    pub id: String,
    pub username: String,
    pub uuid: String,
    pub access_token: String,
    pub account_type: AccountType,
}

File: src/types/account.ts
export type AccountType = "offline" | "microsoft";

export interface MinecraftAccount {
  id: string;
  username: string;
  uuid: string;
  accessToken: string;
  accountType: AccountType;
}

export interface MicrosoftAccountInfo {
  username: string;
  uuid: string;
  tokenExpiresAt: string;
}

export interface DeviceCodeInfo {
  deviceCode: string;
  userCode: string;
  verificationUri: string;
  expiresIn: number;
  interval: number;
}
